### ISSUE TITLE
[BUG] Failures not filtered from data frame when calling fit_surrogate

### ISSUE THREAD

**Issue by Deathn0t** [deephyper/deephyper#181]
**Describe the bug**

Failures should be filtered otherwise the call fails. The same idea can be followed as done in `fit_generative_model`:

```pyton
# check single or multiple objectives
        hp_cols = [k for k in df.columns if "p:" == k[:2]]
        if "objective" in df.columns:
            # filter failures
            if pd.api.types.is_string_dtype(df.objective):
                df = df[~df.objective.str.startswith("F")]
                df.objective = df.objective.astype(float)

            q_val = np.quantile(df.objective.values, q)
            req_df = df.loc[df["objective"] > q_val]
        else:
            # filter failures
            objcol = list(df.filter(regex=r"^objective_\d+$").columns)
            for col in objcol:
                if pd.api.types.is_string_dtype(df[col]):
                    df = df[~df[col].str.startswith("F")]
                    df[col] = df[col].astype(float)
```

**Comment by Deathn0t** [comment]
Related to https://github.com/deephyper/deephyper/issues/191

**Comment by Deathn0t** [comment]
Resolved in 82e678cbedba6d164d78900185fc94bcb6ad727c
